compiler/src/iree/compiler/Codegen/Transforms/Transforms.cpp

```diff
@@ -758,6 +758,41 @@ void packAllocs(OpBuilder &builder, func::FuncOp funcOp,
   }
 }
 
+LogicalResult
+tileLinalgOpsWithFilter(func::FuncOp funcOp, scf::SCFTilingOptions options,
+                        IREE::LinalgExt::LinalgTransformationFilter filter) {
+  IRRewriter rewriter(funcOp.getContext());
+  SmallVector<Operation *> candidates;
+  funcOp.walk([&](linalg::LinalgOp op) {
+    if (succeeded(filter.checkAndNotify(rewriter, op))) {
+      candidates.push_back(op);
+    }
+  });
+
+  for (auto op : candidates) {
+    auto target = cast<TilingInterface>(op);
+
+    // tileUsingSCFForOp requires the op is not able to tile op with no
+    // iteration domain. Skip the tiling on the op. Otherwise it returns a
+    // failure.
+    if (target.getLoopIteratorTypes().empty()) {
+      continue;
+    }
```

Comment:
We should probably update the logic, like [this](https://github.com/llvm/llvm-project/pull/75410/commits/c07f7e1c5c6f8bbc7189e96096004d39a0a1aa3f). We can return the cloned op, so it does not return a failure. This can be a follow-up.

https://github.com/llvm/llvm-project/blob/f4b5be1ecdc85ca4257b739afb8d57e23c7a8030/mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp#L283-L289